versionology/version_fetcher.go

```diff
@@ -0,0 +1,62 @@
+package versionology
+
+import (
+	"sort"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/joshuatcasey/collections"
+)
+
+// VersionFetcher exists to allow buildpack authors to use a type of their own choosing when passing structs
+// in and out of libdependency APIs.
+type VersionFetcher interface {
+	Version() *semver.Version
+}
```

Comment:
I think I'm in the most agreement with this sentiment regarding `versionology.VersionFetcher`

> This is a confusing set of type names when paired with the libdependency.VersionFetcherFunc. Arguably the VersionFetcherFunc is a VersionFetcher and the VersionFetcher is a Version. I find that I've got version.Version().String() in a couple of places and that feels odd.

> I still don't fully grasp why the versionology package exists. In my alternative implementation, I pass around strings with the contract that they be parsable SemVer versions.

Specifically around having to use `version.Version().String()` and the layers wrapped around a version feel confusing as a user experience. Throughout this I have wanted to see the `versionology` package simplified as much as possible as its the part of the library I get confused about the most. I'd love to see if there's a way to simplify the implementations of the retrieval package without needing such complexity around versions themselves